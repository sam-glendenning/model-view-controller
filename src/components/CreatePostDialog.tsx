import {
  Dialog,
  DialogTitle,
  DialogContent,
  Box,
  TextField,
  DialogActions,
  Button,
} from '@mui/material';
import React, { useCallback } from 'react';
import { useCreatePostController } from '@/controllers/useCreatePostController';

interface CreatePostDialogProps {
  isOpen: boolean;
  onClose: () => void;
  onSuccess?: (message: string) => void;
  onError?: (message: string) => void;
}

export const CreatePostDialog: React.FC<CreatePostDialogProps> = ({
  isOpen,
  onClose,
  onSuccess,
  onError,
}) => {
  const {
    formData,
    isSubmitDisabled,
    handleClose,
    updateTitle,
    updateBody,
    updateUserId,
    submitForm,
  } = useCreatePostController({
    onClose,
    onSuccess,
    onError,
  });

  const handleSubmit = useCallback(async () => {
    await submitForm();
  }, [submitForm]);

  return (
    <Dialog open={isOpen} onClose={handleClose} maxWidth="sm" fullWidth>
      <DialogTitle>Create New Post</DialogTitle>
      <DialogContent>
        <Box sx={{ pt: 1 }}>
          <TextField
            autoFocus
            margin="dense"
            label="Title"
            type="text"
            fullWidth
            variant="outlined"
            value={formData.title}
            onChange={e => updateTitle(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="Content"
            type="text"
            fullWidth
            multiline
            rows={4}
            variant="outlined"
            value={formData.body}
            onChange={e => updateBody(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="User ID"
            type="number"
            fullWidth
            variant="outlined"
            value={formData.userId}
            onChange={e => updateUserId(parseInt(e.target.value) ?? 1)}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose}>Cancel</Button>
        <Button
          onClick={handleSubmit}
          variant="contained"
          disabled={isSubmitDisabled}
        >
          Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};
