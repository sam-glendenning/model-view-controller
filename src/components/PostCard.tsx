import React from 'react';
import {
  Card,
  CardContent,
  CardActions,
  Typography,
  Skeleton,
  Box,
  IconButton,
} from '@mui/material';
import { Edit, Delete, Visibility } from '@mui/icons-material';
import { Post } from '@/types';

interface PostCardProps {
  post?: Post;
  isLoading?: boolean;
  onView?: (post: Post) => void;
  onEdit?: (post: Post) => void;
  onDelete?: (post: Post) => void;
  showActions?: boolean;
}

export const PostCard: React.FC<PostCardProps> = ({
  post,
  isLoading = false,
  onView,
  onEdit,
  onDelete,
  showActions = true,
}) => {
  if (isLoading) {
    return (
      <Card sx={{ minHeight: 200, mb: 2 }}>
        <CardContent>
          <Skeleton variant="text" width="60%" height={32} />
          <Skeleton variant="text" width="100%" height={20} sx={{ mt: 1 }} />
          <Skeleton variant="text" width="100%" height={20} />
          <Skeleton variant="text" width="80%" height={20} />
        </CardContent>
        {showActions && (
          <CardActions>
            <Skeleton variant="rectangular" width={80} height={36} />
            <Skeleton variant="rectangular" width={80} height={36} />
            <Skeleton variant="rectangular" width={80} height={36} />
          </CardActions>
        )}
      </Card>
    );
  }

  if (!post) {
    return (
      <Card sx={{ minHeight: 200, mb: 2 }}>
        <CardContent>
          <Typography variant="h6" color="text.secondary">
            No post data available
          </Typography>
        </CardContent>
      </Card>
    );
  }

  return (
    <Card sx={{ minHeight: 200, mb: 2, transition: 'elevation 0.2s' }} elevation={2}>
      <CardContent>
        <Typography
          variant="h6"
          component="h3"
          gutterBottom
          sx={{
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            display: '-webkit-box',
            WebkitLineClamp: 2,
            WebkitBoxOrient: 'vertical',
          }}
        >
          {post.title}
        </Typography>
        <Typography
          variant="body2"
          color="text.secondary"
          sx={{
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            display: '-webkit-box',
            WebkitLineClamp: 3,
            WebkitBoxOrient: 'vertical',
            lineHeight: 1.4,
          }}
        >
          {post.body}
        </Typography>
        <Box sx={{ mt: 2 }}>
          <Typography variant="caption" color="text.secondary">
            Post ID: {post.id} | User ID: {post.userId}
          </Typography>
        </Box>
      </CardContent>
      {showActions && (
        <CardActions sx={{ justifyContent: 'space-between', px: 2 }}>
          <Box>
            {onView && (
              <IconButton
                size="small"
                onClick={() => onView(post)}
                aria-label="view post"
                color="primary"
              >
                <Visibility />
              </IconButton>
            )}
            {onEdit && (
              <IconButton
                size="small"
                onClick={() => onEdit(post)}
                aria-label="edit post"
                color="secondary"
              >
                <Edit />
              </IconButton>
            )}
            {onDelete && (
              <IconButton
                size="small"
                onClick={() => onDelete(post)}
                aria-label="delete post"
                color="error"
              >
                <Delete />
              </IconButton>
            )}
          </Box>
        </CardActions>
      )}
    </Card>
  );
};
