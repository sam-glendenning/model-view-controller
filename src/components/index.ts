export { PostCard } from './PostCard';
export { UserCard } from './UserCard';
export { PostsList } from './PostsList';
export { UsersList } from './UsersList';
