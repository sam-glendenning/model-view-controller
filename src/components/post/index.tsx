import React from 'react';
import { useUpdatePostController } from '@/controllers/useUpdatePostController';
import { useDeletePostController } from '@/controllers/useDeletePostController';
import { PostViewComponent } from './view/PostViewComponent';
import { PostEditComponent } from './edit/PostEditComponent';
import { DeletePostDialog } from './delete/DeletePostDialog';
import type { Post } from '@/types';

interface PostComponentProps {
  postData: Post;
  onPostDeleted?: (deletedPost: Post) => void;
  onPostUpdated?: (updatedPost: Post) => void;
}

const PostComponent: React.FC<PostComponentProps> = ({
  postData,
  onPostDeleted,
  onPostUpdated,
}) => {
  const {
    isEditingPost,
    formData,
    isUpdating,
    isSaveButtonDisabled: isUpdateSaveButtonDisabled,
    startEditing,
    cancelEditing,
    updateTitle,
    updateBody,
    updateUserId,
    updatePost,
  } = useUpdatePostController({ postData, onPostUpdated });

  const {
    isDeletePostDialogOpen,
    isDeleting,
    showDeletePostDialog,
    hideDeletePostDialog,
    confirmDelete,
  } = useDeletePostController({
    postData,
    onPostDeleted,
  });

  return (
    <div>
      {isEditingPost ? (
        <PostEditComponent
          postData={formData}
          onTitleChange={updateTitle}
          onBodyChange={updateBody}
          onUserIdChange={updateUserId}
          onSave={updatePost}
          onCancel={cancelEditing}
          onDelete={showDeletePostDialog}
          isSaveButtonDisabled={isUpdateSaveButtonDisabled}
          isUpdating={isUpdating}
          isDeleting={isDeleting}
        />
      ) : (
        <PostViewComponent
          postData={postData}
          onEditClick={startEditing}
          onDeleteClick={showDeletePostDialog}
          isDeleting={isDeleting}
        />
      )}
      <DeletePostDialog
        open={isDeletePostDialogOpen}
        postData={postData}
        confirmPostDelete={confirmDelete}
        cancelPostDelete={hideDeletePostDialog}
        isDeleting={isDeleting}
      />
    </div>
  );
};

export default PostComponent;
