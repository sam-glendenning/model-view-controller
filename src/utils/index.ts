export * from './validation';
export * from './errorHandling';
