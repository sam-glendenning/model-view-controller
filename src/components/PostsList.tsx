import React from 'react';
import { Grid, Typography, Box, Alert, CircularProgress } from '@mui/material';
import { PostCard } from './PostCard';
import { Post } from '@/types';

interface PostsListProps {
  posts?: Post[];
  isLoading?: boolean;
  error?: string | null;
  title?: string;
  onViewPost?: (post: Post) => void;
  onEditPost?: (post: Post) => void;
  onDeletePost?: (post: Post) => void;
  showActions?: boolean;
  emptyMessage?: string;
}

export const PostsList: React.FC<PostsListProps> = ({
  posts = [],
  isLoading = false,
  error = null,
  title = 'Posts',
  onViewPost,
  onEditPost,
  onDeletePost,
  showActions = true,
  emptyMessage = 'No posts found',
}) => {
  if (error) {
    return (
      <Box sx={{ mb: 3 }}>
        <Typography variant="h5" gutterBottom>
          {title}
        </Typography>
        <Alert severity="error">
          Error loading posts: {error}
        </Alert>
      </Box>
    );
  }

  const renderLoadingSkeletons = () => {
    return Array.from({ length: 6 }, (_, index) => (
      <Grid item xs={12} sm={6} md={4} key={`skeleton-${index}`}>
        <PostCard isLoading={true} showActions={showActions} />
      </Grid>
    ));
  };

  return (
    <Box sx={{ mb: 3 }}>
      <Box sx={{ display: 'flex', alignItems: 'center', mb: 3 }}>
        <Typography variant="h5" gutterBottom sx={{ mb: 0, flexGrow: 1 }}>
          {title}
        </Typography>
        {isLoading && (
          <CircularProgress size={24} sx={{ ml: 2 }} />
        )}
      </Box>

      <Grid container spacing={2}>
        {isLoading && renderLoadingSkeletons()}
        
        {!isLoading && posts.length === 0 && (
          <Grid item xs={12}>
            <Alert severity="info">
              {emptyMessage}
            </Alert>
          </Grid>
        )}
        
        {!isLoading && posts.map((post) => (
          <Grid item xs={12} sm={6} md={4} key={post.id}>
            <PostCard
              post={post}
              onView={onViewPost}
              onEdit={onEditPost}
              onDelete={onDeletePost}
              showActions={showActions}
            />
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};
