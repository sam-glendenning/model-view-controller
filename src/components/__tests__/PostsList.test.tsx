import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PostsList } from '@/components/PostsList';
import { render } from '@/test/utils';
import { mockPosts } from '@/test/mocks/handlers';

describe('PostsList', () => {
  const mockHandlers = {
    onViewPost: jest.fn(),
    onEditPost: jest.fn(),
    onDeletePost: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Snapshot Tests', () => {
    it('renders posts list with data correctly', () => {
      const { container } = render(
        <PostsList
          posts={mockPosts}
          isLoading={false}
          title="Test Posts"
          onViewPost={mockHandlers.onViewPost}
          onEditPost={mockHandlers.onEditPost}
          onDeletePost={mockHandlers.onDeletePost}
          showActions={true}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders posts list without actions correctly', () => {
      const { container } = render(
        <PostsList
          posts={mockPosts}
          isLoading={false}
          title="Test Posts"
          showActions={false}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Loading State', () => {
    it('displays loading skeletons when isLoading is true', () => {
      render(
        <PostsList
          posts={[]}
          isLoading={true}
          title="Loading Posts"
        />
      );
      
      expect(screen.getByText('Loading Posts')).toBeInTheDocument();
      expect(screen.getByRole('progressbar')).toBeInTheDocument();
      
      // Should render skeleton cards
      const skeletons = document.querySelectorAll('.MuiSkeleton-root');
      expect(skeletons.length).toBeGreaterThan(0);
    });
  });

  describe('Error State', () => {
    it('displays error message when error is provided', () => {
      const errorMessage = 'Failed to load posts';
      render(
        <PostsList
          posts={[]}
          isLoading={false}
          error={errorMessage}
          title="Posts"
        />
      );
      
      expect(screen.getByText('Posts')).toBeInTheDocument();
      expect(screen.getByText(`Error loading posts: ${errorMessage}`)).toBeInTheDocument();
    });
  });

  describe('Empty State', () => {
    it('displays default empty message when no posts', () => {
      render(
        <PostsList
          posts={[]}
          isLoading={false}
          title="No Posts"
        />
      );
      
      expect(screen.getByText('No Posts')).toBeInTheDocument();
      expect(screen.getByText('No posts found')).toBeInTheDocument();
    });

    it('displays custom empty message', () => {
      const customMessage = 'No posts available for this user';
      render(
        <PostsList
          posts={[]}
          isLoading={false}
          title="User Posts"
          emptyMessage={customMessage}
        />
      );
      
      expect(screen.getByText(customMessage)).toBeInTheDocument();
    });
  });

  describe('Content Display', () => {
    it('displays all posts', () => {
      render(
        <PostsList
          posts={mockPosts}
          isLoading={false}
          title="All Posts"
        />
      );
      
      mockPosts.forEach(post => {
        expect(screen.getByText(post.title)).toBeInTheDocument();
      });
    });

    it('displays custom title', () => {
      const customTitle = 'My Custom Posts';
      render(
        <PostsList
          posts={mockPosts}
          isLoading={false}
          title={customTitle}
        />
      );
      
      expect(screen.getByText(customTitle)).toBeInTheDocument();
    });
  });

  describe('Action Handling', () => {
    it('passes handlers to post cards when showActions is true', async () => {
      const user = userEvent.setup();
      render(
        <PostsList
          posts={[mockPosts[0]]}
          isLoading={false}
          onViewPost={mockHandlers.onViewPost}
          onEditPost={mockHandlers.onEditPost}
          onDeletePost={mockHandlers.onDeletePost}
          showActions={true}
        />
      );
      
      // Click view button on first post
      const viewButton = screen.getByLabelText('view post');
      await user.click(viewButton);
      
      expect(mockHandlers.onViewPost).toHaveBeenCalledWith(mockPosts[0]);
    });

    it('does not show action buttons when showActions is false', () => {
      render(
        <PostsList
          posts={[mockPosts[0]]}
          isLoading={false}
          showActions={false}
        />
      );
      
      expect(screen.queryByLabelText('view post')).not.toBeInTheDocument();
      expect(screen.queryByLabelText('edit post')).not.toBeInTheDocument();
      expect(screen.queryByLabelText('delete post')).not.toBeInTheDocument();
    });
  });

  describe('Responsive Layout', () => {
    it('renders posts in grid layout', () => {
      render(
        <PostsList
          posts={mockPosts}
          isLoading={false}
          title="Grid Posts"
        />
      );
      
      // Check that posts are rendered in a grid container
      const gridContainer = document.querySelector('[class*="MuiGrid-container"]');
      expect(gridContainer).toBeInTheDocument();
    });
  });
});
