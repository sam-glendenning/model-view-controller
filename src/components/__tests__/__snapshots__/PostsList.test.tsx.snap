// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList Snapshot Tests renders posts list with data correctly 1`] = `
<div
  class="MuiBox-root css-i3pbo"
>
  <div
    class="MuiBox-root css-qr12fg"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13omkd-MuiTypography-root"
    >
      Test Posts
    </h5>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Test Post Title
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            This is a test post body with some content to display.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              1
               | User ID: 
              1
            </span>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-p5c1gw-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-label="view post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="edit post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall css-1hxb263-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="delete post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1mk6yu1-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Another Test Post
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            This is another test post with different content.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              2
               | User ID: 
              1
            </span>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-p5c1gw-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-label="view post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="edit post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall css-1hxb263-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="delete post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1mk6yu1-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Jane's Post
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            A post written by Jane Smith about various topics.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              3
               | User ID: 
              2
            </span>
          </div>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-p5c1gw-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-label="view post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5M12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5m0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="edit post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall css-1hxb263-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="EditIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="delete post"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1mk6yu1-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostsList Snapshot Tests renders posts list without actions correctly 1`] = `
<div
  class="MuiBox-root css-i3pbo"
>
  <div
    class="MuiBox-root css-qr12fg"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-13omkd-MuiTypography-root"
    >
      Test Posts
    </h5>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Test Post Title
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            This is a test post body with some content to display.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              1
               | User ID: 
              1
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Another Test Post
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            This is another test post with different content.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              2
               | User ID: 
              1
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-11l5t4l-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-rpm3sl-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h3
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1q6nwsw-MuiTypography-root"
          >
            Jane's Post
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body2 css-yb0dix-MuiTypography-root"
          >
            A post written by Jane Smith about various topics.
          </p>
          <div
            class="MuiBox-root css-1yuhvjn"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-ol7wxs-MuiTypography-root"
            >
              Post ID: 
              3
               | User ID: 
              2
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
